ls8/cpu.py

```diff
@@ -62,4 +91,73 @@ def trace(self):
 
     def run(self):
         """Run the CPU."""
-        pass
+
+        LDI = 0b10000010
+        HLT = 0b00000001
+        MUL = 0b10100010
+        ADD = 0b10100000
+        DIV = 0b10100011
+        SUB = 0b10100001
+        PUSH = 0b01000101
+        POP = 0b01000110
+        CALL = 0b01010000
+        RET = 0b00010001
+        NOP = 0b00000000
+        PRN = 0b01000111
+
+        running = True
+        self.reg[self.sp] = 0b11111111  # 255
+        while running:
+            # It needs to read the memory address that's stored in register `PC`, and store
+            # that result in `IR`, the _Instruction Register_. This can just be a local
+            # variable in `run()`.
+            IR = self.ram[self.pc]
+            operand_a = self.ram_read(self.pc + 1)
+            operand_b = self.ram_read(self.pc + 2)
+            if IR == LDI:
+                self.reg[operand_a] = operand_b
+                self.pc += 3
+            elif IR == PRN:
+                print(self.reg[operand_a])
+                self.pc += 2
+            elif IR == MUL:
+                self.alu('MUL', operand_a, operand_b)
+                self.pc += 3
+            elif IR == ADD:
+                self.alu('ADD', operand_a, operand_b)
+                self.pc += 3
+            elif IR == DIV:
+                self.alu('DIV', operand_a, operand_b)
+                self.pc += 3
+            elif IR == SUB:
+                self.alu('SUB', operand_a, operand_b)
+                self.pc += 3
+            elif IR == NOP:
+               pass
```

Comment:
nit picking: it would be a good idea to print a message here to let the user know that the instruction was recognized but no action is set up.

Reply:
Ok I will add that, then. Thanks.